src/main/java/seedu/lifeasier/parser/Parser.java

```diff
@@ -883,6 +894,122 @@ private boolean isNumeric(String input) {
+        //Check if there are 4 consecutive numbers in sequence
+        if (numberCount != 4) {
+            logger.log(Level.INFO, "Invalid numeric code length");
+            return false;
+        }
+
+        //Checking for existence of postfix
+        int totalCharacterCount = letterCount + numberCount;
+        boolean isFinalLength = totalCharacterCount == moduleCodeElements.length;
+        if (isFinalLength) {
+            logger.log(Level.INFO, "Check done - Valid module code");
+            return true;
+        }
+
+        //Possible postfix, check for valid postfix
+        int numberOfPostfixCharacters = moduleCodeElements.length - totalCharacterCount;
+        if (numberOfPostfixCharacters > MAXIMUM_POSTFIX_CHARACTER) {
+            logger.log(Level.INFO, "Invalid postfix length");
+            return false;
+        }
+
+        logger.log(Level.INFO, "Checking for module postfix letter");
+        int lastIndex = moduleCodeElements.length - 1;
+        if (Character.isLetter(moduleCodeElements[lastIndex])) {
+            logger.log(Level.INFO, "Check done - Valid module code");
+            return true;
+        } else {
+            logger.log(Level.INFO, "Invalid module code postfix");
+            return false;
+        }
+    }
+
+    /**
+     * Counts the number of consecutive numbers for the module code.
+     *
+     * @param moduleCodeElements Array of characters from module code.
+     * @param letterCount Starting index from the number of prefix characters.
+     * @return The number of consecutive numbers.
+     */
+    protected int getNumberCount(char[] moduleCodeElements, int letterCount) {
+        assert letterCount != 0 : "LetterCount should not be 0 at this state";
+        logger.log(Level.INFO, "Start numeric code count");
+        int numberCount = 0;
+        for (int i = letterCount; i < moduleCodeElements.length; i++) {
+            if (Character.isDigit(moduleCodeElements[i])) {
+                numberCount++;
+            } else {
+                break;
+            }
+        }
+        return numberCount;
+    }
+
+    /**
+     * Counts the number of consecutive letters for the module code prefix within the first 4 elements of array.
+     *
+     * @param moduleCodeElements Array of characters from module code.
+     * @return The number of consecutive letters.
+     */
+    protected int getLetterCount(char[] moduleCodeElements) {
+        logger.log(Level.INFO, "Start prefix letter count");
+        int letterCount = 0;
+        for (int i = 0; i < 4; i++) {
+            if (Character.isLetter(moduleCodeElements[i])) {
+                letterCount++;
+            } else {
+                break;
+            }
+        }
+        return letterCount;
+    }
+
+    /**
+     * Repeatedly prompts for valid module code until a valid one is input.
+     *
+     * @return Valid module code.
+     */
+    public String getValidModuleCode() {
+        Ui ui = new Ui();
```

Comment:
Maybe we could find a way to do this that doesn't involve instantiating another Ui object?

Reply:
I believe a ui object is already passed into parser when we call parseCommand.

![image](https://user-images.githubusercontent.com/47075579/98248282-af8e2b80-1faf-11eb-9d0c-92a70561c892.png)